dcc-portal-ui/app/scripts/pathwayviewer/js/directive.js

```diff
@@ -407,7 +408,7 @@
 
         // Render legend last to ensure all dependencies are initialized. Timeout of 0 does not work in firefox.
         $scope.$on(PathwaysConstants.EVENTS.MODEL_READY_EVENT, function() {
-            controller.renderLegend(270, 671);
+            // TO-DO - finish separation of legend's creation and insertion logic
```

Comment:
No -- the legend logic is separated and is displaying, but it is rendered and appended at a different point in the code